tests/unit/core/test_tracking.js

```diff
@@ -353,4 +357,194 @@ describe(__filename, () => {
+    });
+
+    it('should return null from the init promise if hctEnabled is false', async () => {
+      const tracking = createTracking({
+        _config: stubConfig({ hctEnabled: false }),
+      });
+      const hctLib = await tracking.hctInitPromise;
+      expect(hctLib).toEqual(null);
+    });
+
+    it('should return hct object from the init promise if hctEnabled is true', async () => {
+      const tracking = createTracking({
+        _config: stubConfig({ hctEnabled: true }),
+      });
+      const hctLib = await tracking.hctInitPromise;
+      expect(hctLib).toHaveProperty('canUpload');
+      expect(hctLib).toHaveProperty('initPromise');
+      expect(hctLib).toHaveProperty('recordEvent');
+      expect(hctLib).toHaveProperty('registerEvents');
+    });
+
+    it('should call registerEvents if hctEnabled is true', async () => {
+      const tracking = createTracking({
+        _config: stubConfig({ hctEnabled: true }),
+      });
+      await tracking.hctInitPromise;
+      sinon.assert.calledOnce(registerEventsSpy);
+    });
+  });
+
+  describe('Hybrid Content Telemetry Events', () => {
+    let importStub;
+    let canUploadStub;
+    let recordEventSpy;
+    const trackingData = {
+      method: INSTALL_EXTENSION_CATEGORY,
+      object: TRACKING_TYPE_EXTENSION,
+      value: 'value',
+    };
+
+    beforeEach(() => {
+      importStub = sinon.stub(hct, 'initPromise').callsFake(() => {
+        return Promise.resolve(hct);
+      });
+      canUploadStub = sinon.stub(hct, 'canUpload');
+      recordEventSpy = sinon.spy(hct, 'recordEvent');
+    });
+
+    afterEach(() => {
+      importStub.restore();
+      canUploadStub.restore();
+      recordEventSpy.restore();
+    });
+
+    it('should not call recordEvent if canUpload returns false', async () => {
+      canUploadStub.callsFake(() => false);
+      const tracking = createTracking({
+        _config: stubConfig({ hctEnabled: true }),
+      });
+
+      await tracking._hct(trackingData);
+      sinon.assert.notCalled(recordEventSpy);
+    });
+
+    it('should call recordEvent if canUpload is true', async () => {
+      canUploadStub.callsFake(() => true);
+      const tracking = createTracking({
+        _config: stubConfig({ hctEnabled: true }),
+      });
+
+      await tracking._hct(trackingData);
+      sinon.assert.calledOnce(recordEventSpy);
+      sinon.assert.calledWith(
+        recordEventSpy,
+        'disco.interaction',
+        filterIdentifier(INSTALL_EXTENSION_CATEGORY),
+        filterIdentifier(TRACKING_TYPE_EXTENSION),
+        'value',
+      );
+    });
+
+    it('should throw if unregistered method is used', () => {
+      canUploadStub.callsFake(() => true);
+      const tracking = createTracking({
+        _config: stubConfig({ hctEnabled: true }),
+      });
+      const badTrackingData = Object.assign({}, trackingData, {
+        method: 'failMethod',
+      });
+      return expect(tracking._hct(badTrackingData)).rejects.toThrow(
```

Comment:
We usually don't test the invariants. That's why we use this lib (to stop writing such test cases). Invariant is not used in production.

Reply:
Ah ok, makes sense 👍